packages/ui5-proxy-middleware/src/base/utils.ts

```diff
@@ -204,20 +204,21 @@ export async function getUI5VersionFromManifest(args: string[]): Promise<string
  * @returns The UI5 version with which the application will be started
  */
 export async function resolveUI5Version(version?: string, log?: ToolsLogger): Promise<string> {
```

Comment:
It took me a moment to fully understand what exactly the function assigns when and it seems that some conditions are checked multiple times. If I understood it correctly then we use the version in the following order:
1. env variable
2. input to the function
3. minVersion from manifest

If that is the case, I would suggest to restructure the code like
```
    let ui5Version: string;
    let ui5VersionInfo: string;
    let ui5VersionLocation: string;
	
	if (process.env.FIORI_TOOLS_UI5_VERSION || process.env.FIORI_TOOLS_UI5_VERSION === '') {
		ui5Version = process.env.FIORI_TOOLS_UI5_VERSION;
		ui5VersionLocation = 'CLI arguments / Run configuration';
	} else if (version) {
		ui5Version = version;
		ui5VersionLocation = getYamlFile(process.argv);
	} else {
		const minUI5Version = await getUI5VersionFromManifest(process.argv);
		ui5Version = minUI5Version && isNaN(parseFloat(minUI5Version)) ? minUI5Version : '';
		ui5VersionLocation =  'manifest.json';
	}
    
    if (log) {
        ui5VersionInfo = ui5Version ? ui5Version : 'latest';
        log.info(t('info.ui5VersionSource', { version: ui5VersionInfo, source: ui5VersionLocation }));
    }
    return ui5Version;
```

Less condition checks and the semantic is easier to grasp from the code structure (at least for me).

Reply:
Thanks @tobiasqueck. Done with https://github.com/SAP/open-ux-tools/pull/490/commits/5d35da0b4a9269c4ce60cdb09ae0cb34ae547a56. I did only 2 small adjustments.

Reply:
Thanks (and thanks for fixing my proposal that was ignoring `version = null`)